### PR TITLE
docs: add diagram and restructure for clarity

### PR DESCRIPTION
**Pull request by robszumski**
I thought a diagram would help make the process clearer and we can re-use it in a blog post or any other documentation.

I restructured some of the sentences into the sections where appropriate.

### CODE REVIEW

File: Documentation/allow_legacy_mode.md [hunk 1]
```diff
@@ -64,19 +84,17 @@ Detect the cluster has been upgraded to internal API v2. Exit now.
...
+#### 3. Check the Cluster Health
 
-### Unavailable window
+After the upgrade process, you can run the health check again to verify the upgrade. If the cluster is unhealthy or there is an unhealthy member, please refer to start [failure recovery](#failure-recovery).
 
-etcd cluster will be unavailable between the time that the upgrade command is sent and the time that each etcd process is upgraded and restarted.
+### Downgrade
 
-Usually, the window should be shorter than 1 minute for clusters storing up to 50 MB.
+If the upgrading fails due to disk/network issues, you still can restart the upgrading process manually. However, once you upgrade etcd to internal API v2, you CANNOT downgrade it back to internal API v1. If you want to downgrade etcd in the future, please backup your v1 data dir beforehand.
```
Comment (yichengq) [inline Documentation/allow_legacy_mode.md]: s/internal API v2/internal v2 protocol/

**Comment by yichengq** [comment]
lgtm

**Comment by xiang90** [comment]
lgtm

File: Documentation/allow_legacy_mode.md (hunk 1 above)
Comment (robszumski) [inline Documentation/allow_legacy_mode.md]: I'll fix this, but FWIW, there is output from etcd that says API:

```
Detect the cluster has been upgraded to internal API v2. Exit now.
```

**Pull request merged**